listenbrainz/webserver/views/user_timeline_event_api.py

```diff
@@ -611,6 +640,119 @@ def create_personal_recommendation_event(user_name):
+                non_followers.append(follower)
+        if non_followers:
+            raise APIBadRequest(
+                f"You cannot recommend tracks to non-followers! These people don't follow you {str(non_followers)}")
+        event = db_user_timeline_event.create_personal_recommendation_event(
+            db_conn, user['id'], metadata)
+    except pydantic.ValidationError as e:
+        raise APIBadRequest(f"Invalid metadata: {str(e)}")
+    except DatabaseException:
+        raise APIInternalServerError("Something went wrong, please try again.")
+
+    event_data = event.dict()
+    event_data['created'] = int(event_data['created'].timestamp())
+    event_data['event_type'] = event_data['event_type'].value
+    return jsonify(event_data)
+
+
+@user_timeline_event_api_bp.post('/user/<user_name>/timeline-event/create/thanks')
+@crossdomain
+@ratelimit()
+def create_thanks_event(user_name):
+    '''
+    Make the user recommend a recording to their followers. The request should post
+    the following data about the recording being recommended (either one of recording_msid
+    or recording_mbid is sufficient), and also the list of followers getting recommended:
+
+    .. code-block:: json
+
+        {
+            "metadata": {
+                "recording_msid": "<The MessyBrainz ID of the recording, optional>",
+                "recording_mbid": "<The MusicBrainz ID of the recording>",
+                "users": [], // "<usernames of the persons you want to recommend to, required>"
+                "blurb_content": "<String containing personalized recommendation>"
+            }
+        }
+
+    :reqheader Authorization: Token <user token>
+    :reqheader Content-Type: *application/json*
+    :statuscode 200: Successful query, recording has been recommended!
+    :statuscode 400: Bad request, check ``response['error']`` for more details.
+    :statuscode 401: Unauthorized, you do not have permissions to recommend
+        personal recordings on the behalf of this user
+    :statuscode 403: Forbidden, you do not have permissions to recommend
+    :statuscode 404: User not found
+    :resheader Content-Type: *application/json*
+    '''
+
+    user = validate_auth_header()
+
+    if user_name != user['musicbrainz_id']:
+        raise APIForbidden(
+            "You don't have permissions to hide events from this user's timeline.")
+
+    try:
+        data = orjson.loads(request.get_data())
+    except (ValueError, KeyError) as e:
+        raise APIBadRequest(f"Invalid JSON: {str(e)}")
+
+    if 'metadata' not in data:
+        raise APIBadRequest("JSON document must contain metadata", data)
+
+    metadata = data["metadata"]
+
+    if 'original_event_type' not in metadata or 'original_event_id' not in metadata:
+        raise APIBadRequest(
+            "Metadata must contain both original_event_type and original_event_id", metadata)
+
+    try:
+        validate = ThanksMetadata(**metadata)
```

Comment:
Main issue I had to resolve is that you are not using the class created here when you send it to `db_user_timeline_event.create_thanks_event` which results in some errors.
Overwriting the metadata variable here resolves that issue
```suggestion
        metadata = ThanksMetadata(**metadata)
```

Reply:
I believe there were a couple of issues resulting from not using the expected class, but you possibly fixed one of them (the UserTimelineEventMetadata union type did not contain the new ThanksMetadata)

If I test now with latest commits and add a logger call to log the error, this is what I get:
```
2025-01-28 15:58:05,802 listenbrainz.webserver ERROR    'dict' object has no attribute 'dict'
```
Thrown by this line: https://github.com/metabrainz/listenbrainz-server/blob/46374f7d66d7d1a21850378cc816a84d53a4e7ed/listenbrainz/db/user_timeline_event.py#L59

Since a `UserTimelineEvent` class is expected in all of these methods, you are likely to run into some issue or other if you pass a dict instead of a class.